### PR TITLE
fix:TextInput textBreakStrategy default value on Android (Paper - old arch)

### PR DESCRIPTION
**Pull request by aswin-s**
## Summary:

This pull request addresses an issue related to the TextInput component. Currently, the default value for the textBreakStrategy prop is set to highQuality on Android platform. This is against the expectation as per RN documentation [here](https://reactnative.dev/docs/textinput#textbreakstrategy-android). 

This PR proposes a fix that updates the default value of textBreakStrategy to simple on Paper. For new arch the value seems to be inherited from ParagraphAttributes and is set to `TextBreakStrategy::HighQuality` by default. Overriding the value and setting it to `Simple` for TextInput seems to be beyond my capabilities. 

fixes https://github.com/facebook/react-native/issues/38244

## Changelog:

[ANDROID] [FIXED] - Set default value of textBreakStrategy to 'simple' on Paper (old-arch)

## Test Plan:

- Run the RNTester App
- Navigate to examples for TextInput
- Ensure `breakStrategy` attribute for EditText components are 'simple' by default. Use LayoutInspector from Android Studio for the same.

### CODE REVIEW

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java [hunk 1]
```diff
@@ -328,19 +328,23 @@ public void receiveCommand(
 
         if (!args.isNull(1)) {
           String text = args.getString(1);
-          reactEditText.maybeSetTextFromJS(getReactTextUpdate(text, mostRecentEventCount));
+          reactEditText.maybeSetTextFromJS(getReactTextUpdate(reactEditText, text, mostRecentEventCount));
```
Comment (philIip) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java:331]: why this change? the break strategy seems to get updated through other pipelines

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputShadowNode.java [hunk 2]
```diff
@@ -53,10 +53,7 @@ public class ReactTextInputShadowNode extends ReactBaseTextShadowNode
   public ReactTextInputShadowNode(
       @Nullable ReactTextViewManagerCallback reactTextViewManagerCallback) {
     super(reactTextViewManagerCallback);
-    mTextBreakStrategy =
-        (Build.VERSION.SDK_INT < Build.VERSION_CODES.M)
-            ? Layout.BREAK_STRATEGY_SIMPLE
-            : Layout.BREAK_STRATEGY_HIGH_QUALITY;
+    mTextBreakStrategy = Layout.BREAK_STRATEGY_SIMPLE;
```
Comment (philIip) [inline packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputShadowNode.java:56]: mmm, tho i understand why you are making this change, i'm concerned that this will break apps that might be assuming HQ.

is there a reason we want to change this at the infra level instead of at the product layer besides inconsistency with website docs?

**Review by philIip — CHANGES_REQUESTED** [review]
is my understanding correct that setting the breakStrategy prop doesn't work in the new arch?

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java (hunk 1 above)
Comment (aswin-s) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java:331]: `getTextUpdate` method was returning a new ReactTextUpdate object with hardcoded break strategy value of `0`. This was value getting copied over with each onChangeText event there by resetting breakStrategy to `0` even when other strategies were used in code.

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputShadowNode.java (hunk 2 above)
Comment (aswin-s) [reply packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputShadowNode.java:56]: Nope. Was just trying to fix the react-native behaviour as per the documentation. Typing in a TextInput with highQuality break strategy is super annoying as the text wrapping keeps on changing with user input.

**Comment by aswin-s** [comment]
@philIip What's your recommendation here? Shouldn't we fix this in react-native itself as the recommended default by Android OS itself is 'simple' for TextInputs. This could obviously be fixed directly in application code, but has to be repeated for each and every TextInput usage. Also unsuspecting devs would simply refer to the react-native documentation and assume that textBreakStrategy is 'simple' by default and needs no modification.

**Comment by aswin-s** [comment]
> @philIip What's your recommendation here? Shouldn't we fix this in react-native itself as the recommended default by Android OS itself is 'simple' for TextInputs. This could obviously be fixed directly in application code, but has to be repeated for each and every TextInput usage. Also unsuspecting devs would simply refer to the react-native documentation and assume that textBreakStrategy is 'simple' by default and needs no modification.

@philIip Still waiting for your response here. We need to fix either the documentation or the default behaviour in code.
